### ISSUE TITLE
GOScoreEvaluation fails because the faulty implementation of the mirror option.

### ISSUE THREAD

**Issue by bobermayer** [jn-goe/gruffi#18]
**Describe the bug**
`GOScoreEvaluation` fails with ` unused argument (mirror = NULL)`. seems like this argument is passed on to `FeaturePlot`

**To Reproduce**

```r
> combined.obj <- GOscoreEvaluation(obj = combined.obj, GO_term = 'GO:0006096',
+                                   save.UMAP = TRUE, new_GO_term_computation = T, 
+                                   clustering = granule.res.4.gruffi, plot.each.gene = F)
[1] "Score.GO.0006096"
[1] "Score.GO.0006096 not found. Need to call GetGOTerms(). set only.draw.plot = FALSE."
[1] "GENE SCORE WILL NOW BE SET / OVERWRITTEN"
[1] "90 gene symbols downloaded via biomaRt::getBM(): SLC2A6 LDHA HTR2A PFKL FOXK1 PGK2 INSR EIF6 PGAM1 KAT2B"
[1] "90 Gene symbols downloaded: SLC2A6 LDHA HTR2A PFKL FOXK1 PGK2 INSR EIF6 PGAM1 KAT2B"
[1] "IntersectWithExpressed()"
[1] "3 genes (of 90 ) are MISSING from the Seurat object: PRXL2C TREX1 IFNG"
[1] "Genes in GO:0006096 are saved under obj@misc$GO$ GO.0006096"
[1] "ENO1"   "MTOR"   "PRKAA2" "PGM1"   "ARNT"   "PKLR"  
[1] "AddGOScore()"
[1] "Trailing '1' in metadata column name is removed. Column name: Score.GO.0006096"
Error in Seurat::FeaturePlot(obj, features = GO.score, min.cutoff = "q05",  : 
  unused argument (mirror = NULL)
> traceback()
3: FeaturePlotSaveGO(obj = obj, GO.score = ScoreName, save.plot = save.UMAP, 
       name_desc = desc, ...)
2: PlotGoTermScores(GO = GO_term, save.UMAP = save.UMAP, obj = obj, 
       desc = description, plot.each.gene = plot.each.gene, mirror = mirror, 
       ...)
1: GOscoreEvaluation(obj = combined.obj, GO_term = go_id_glycolysis, 
       save.UMAP = TRUE, new_GO_term_computation = T, clustering = granule.res.4.gruffi, 
       plot.each.gene = F)
```

<details>
<summary>sessionInfo()</summary>

```{r}
> sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /opt/R/4.3.2/lib/R/lib/libRblas.so 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Berlin
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gruffi_1.2.6          MarkdownReports_4.7.0 MarkdownHelpers_1.0.7 CodeAndRoll2_2.5.6   
 [5] Stringendo_0.6.0      magrittr_2.0.3        tidyr_1.3.1           Matrix_1.6-5         
 [9] igraph_2.0.1.1        cowplot_1.1.3         ggplot2_3.4.4         dplyr_1.1.4          
[13] SeuratObject_4.1.4    Seurat_4.4.0          reticulate_1.35.0    
```

</details>

**Comment by ksarieva** [comment]
Same here with Seurat version 5.0.1.

**Comment by bobermayer** [comment]
removing the `mirror` options from the source code works for me

**Comment by vertesy** [comment]
Hey, thank you for reporting and sorry for the inconvenience. I hope I get to following up on this tomorrow.

**Comment by vertesy** [comment]
I recognize, and this is hopefully already be fixed in `v1.2.7`, the problem was in `PlotGoTermScores()`'s call of `GetGOTerms()` did not specify `mirror`.

```r
obj <- GetGOTerms(obj = obj, GO = GO, web.open = openBrowser, use.ensemble = use.ensemble)
# ...
plot <- FeaturePlotSaveGO(obj = obj, GO.score = ScoreName, save.plot = save.UMAP, name_desc = desc, ...)
```
...and for lack of thus mirror ended up in `FeaturePlotSaveGO()`


Now correctly:

```r
obj <- GetGOTerms(obj = obj, GO = GO, web.open = openBrowser, use.ensemble = use.ensemble, mirror = mirror)
# ...
plot <- FeaturePlotSaveGO(obj = obj, GO.score = ScoreName, save.plot = save.UMAP, name_desc = desc, ...)
```

**Comment by vertesy** [comment]
Of note, as of `v1.3.0`
`GOscoreEvaluation()`  will have the more explicit name of `AssignGranuleAverageScoresFromGOterm()`.
